subprojects/model-core/src/main/java/org/gradle/internal/metaobject/BeanDynamicObject.java

```diff
@@ -369,10 +371,18 @@ public DynamicInvokeResult setProperty(final String name, Object value) {
                         MetaBeanProperty metaBeanProperty = (MetaBeanProperty) property;
                         if (metaBeanProperty.getSetter() == null) {
                             if (metaBeanProperty.getField() == null) {
-                                throw setReadOnlyProperty(name);
+                                Class<?> propertyType = metaBeanProperty.getType();
+                                if (HasConfigurableValue.class.isAssignableFrom(propertyType)) {
+                                    value = propertySetTransformer.transformValue(propertyType, value);
+                                    LazyGroovySupport propertyValue = (LazyGroovySupport) metaBeanProperty.getGetter().invoke(bean, new Object[0]);
```

Comment:
🤔 Does this cast always work?

Reply:
We now check that instance is really a LazyGroovySupport